client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java

```diff
@@ -471,27 +471,36 @@ public void testConcurrentCancellationOnStatementClose()
         Future<?> future;
         try (Connection connection = createConnection()) {
             Statement statement = connection.createStatement();
-            future = executor.submit(() -> statement.execute(sql));
+            future = executor.submit(() -> {
+                try (ResultSet resultSet = statement.executeQuery(sql)) {
+                    //noinspection StatementWithEmptyBody
+                    while (resultSet.next()) {
+                        // consume results
+                    }
+                }
+                return null;
+            });
 
             // Wait for the queries to be started
-            assertEventually(() -> assertThat(listQueryStatuses(sql))
-                    .contains("RUNNING")
-                    .hasSize(1));
+            assertEventually(() -> {
+                assertThatFutureIsBlocked(future);
+                assertThat(listQueryStatuses(sql))
+                        .contains("RUNNING")
+                        .hasSize(1);
+            });
 
             // Closing statement should cancel queries
             statement.close();
 
             // verify that the query was cancelled
-            assertThatThrownBy(future::get)
-                    .hasCauseInstanceOf(SQLException.class)
-                    .hasStackTraceContaining("Error executing query");
+            assertThatThrownBy(future::get);
```

Comment:
missing assertion

Reply:
`assertThatThrownBy` verifies that exception was throw without checking exception details.

Reply:
@ksobolew will it violate https://github.com/trinodb/trino/pull/8209?

@kokosing even if it doesn't, it violates "ignored return value" in "my eyes". Maybe `assertThat(catchThrowable(future::get)).isNotNull()`?

Reply:
This particular method is annotated with `@CanIgnoreReturnValue`, so it shouldn't, but this annotation is AssertJ-specific and I don't know if Error Prone knows about it

Reply:
The internal implementation of `assertThatThrownBy` is `assertThat(catchThrowable(future::get)).hasBeenThrown()` which is nicer than `isNotNull`.

Reply:
@ksobolew i know, but `hasBeenThrown` isn't public

Reply:
Oh. That's sad ☹️